packages/http-client/src/ceramic-http-client.ts

```diff
@@ -63,11 +66,11 @@ export default class CeramicClient implements CeramicApi {
   private readonly _config: CeramicClientConfig
   public readonly _doctypeHandlers: Record<string, DoctypeHandler<Doctype>>
 
-  constructor (apiHost: string = CERAMIC_HOST, config?: CeramicClientConfig) {
-    this._config = Object.assign(DEFAULT_CLIENT_CONFIG, config ? config : {})
+  constructor (apiHost: string = CERAMIC_HOST, config: CeramicClientConfig = {}) {
+    this._config = { ...DEFAULT_CLIENT_CONFIG, ...config }
 
     this._apiUrl = combineURLs(apiHost, API_PATH)
-    this._docmap = {}
+    this._docCache = new DocCache(config.docCacheLimit, true)
```

Comment:
I think we should probably add `cacheDocCommits` to CeramicClientConfig so it can be configured on the http-client side as well.  I actually think the http client probably should not cache doc commits by default, unlike the daemon side.  It makes sense for the daemon to cache commits, since it'll be getting queries from the CAS for documents at specific commits, but the http client won't be, so we should only cache commits if the user explicitly asks us to do so.